File: P1/Ejercicio4/filters/password_length_filter.rb
# filtro que valida que la contraseña tenga una longitud mínima
# implementa la interfaz Filter
class PasswordLengthFilter
  include Filter

  def execute(request)
    # comprueba que la contraseña tenga al menos 8 caracteres
    raise ValidationError, "Contraseña demasiado corta" if request[:password].length < 8
  end
end

File: P2/Ejercicio1_MantenimientoFlutter/lib/filters/PasswordLengthFilter.dart
import '../interfaces/Filter.dart';

// filtro encargado de la longitud de la contraseña (menos de 8 digitos)
class PasswordLengthFilter implements Filter {
  @override
  String? execute(Map<String, String> request) {
    final password = request['password'] ?? '';
    if (password.length < 8) {
      return "Contraseña demasiado corta (mínimo 8 caracteres)";
    }
    return null;
  }
}